dhall/src/Dhall/TypeCheck.hs

```diff
@@ -117,33 +117,34 @@ typeWithA tpa = loop
                 return a
     loop ctx   (Lam x _A  b     ) = do
         _ <- loop ctx _A
-        let ctx' = fmap (Dhall.Core.shift 1 (V x 0)) (Dhall.Context.insert x (Dhall.Core.normalize _A) ctx)
+        let nf_A = Dhall.Core.normalize _A
+        let ctx' = fmap (Dhall.Core.shift 1 (V x 0)) (Dhall.Context.insert x nf_A ctx) -- TODO: Add test case
         _B <- loop ctx' b
-        let p = Pi x _A _B
+        let p = Pi x nf_A _B -- TODO: Add test
```

Comment:
Inferred types should be normalized: FAIL (0.28s)
      *** Failed! Falsifiable (after 54636 tests and 2 shrinks):
      Lam "" (Let "" Nothing NaturalEven Text) (Const Type)
      Got: Right (Pi "" (Let "" Nothing NaturalEven Text) (Const Kind))
      Expected: Right (Pi "" Text (Const Kind))
      Use --quickcheck-replay=737763 to reproduce.